www/index.html.spt

```diff
@@ -17,16 +22,39 @@ teams = website.db.all("""
   ORDER BY ctime DESC
 
 """)
+
 volume = website.db.one("SELECT volume FROM paydays ORDER BY ts_start DESC LIMIT 1", default=0)
 volume = int(round(volume, -2))
 
 tabs = OrderedDict()
+tabs['featured'] = {'teams': []}
 tabs['approved'] = {'teams': []}
 tabs['unreviewed'] = {'teams': []}
 tabs['rejected'] = {'teams': []}
 
-for team in teams:
-    tabs[team.status]['teams'].append(team)
+popular = []
+unpopular = []
+for project in teams:
+    tabs[project.status]['teams'].append(project)
+    if project.status == 'approved':
+      if project.nreceiving_from >= 5:
+        popular.append(project)
+      else:
+        unpopular.append(project)
+
+if len(teams) < 10:
+    tabs.pop('featured')
+else:
+    popular_sample_size = min(len(popular), 7)
+    unpopular_sample_size = min(len(unpopular), 10-popular_sample_size)
+    if popular_sample_size == 0:
+        featured_projects = random.sample(unpopular, unpopular_sample_size)
+    elif unpopular_sample_size == 0:
+        featured_projects = random.sample(popular, min(10, len(popular)))
+    else:
+        featured_projects = random.sample(popular, popular_sample_size) + random.sample(unpopular, unpopular_sample_size)
```

Comment:
Let's go with `test_utils.py` since `utils` seems the natural place to add the function we're talking about. I've added dd0b8ce81863b9e7b8ce75cc4536cba36fd12935 to bring the `test_utils.py` file up to modern standards of organization (one class per function). Hopefully that gives you enough to go on to add tests for the new function